apps/web/src/views/BuyCrypto/components/GetQuotesButton.tsx

```diff
@@ -2,37 +2,29 @@ import ConnectWalletButton from 'components/ConnectWalletButton'
 
 import { AutoColumn, CircleLoader, Flex, Text } from '@pancakeswap/uikit'
 import { CommitButton } from 'components/CommitButton'
-import { Dispatch, ReactNode, useCallback, useState } from 'react'
+import { Dispatch, ReactNode, useCallback } from 'react'
 import { useAccount } from 'wagmi'
 import { useTranslation } from '@pancakeswap/localization'
 import { SetStateAction } from 'jotai'
 import { CryptoFormView } from 'views/BuyCrypto/types'
+import { useMutation } from '@tanstack/react-query'
 
 interface GetQuotesButtonProps {
-  modalView: CryptoFormView
   errorText: string | undefined
   setModalView: Dispatch<SetStateAction<CryptoFormView>>
   fetchQuotes: () => Promise<void>
 }
 
-export default function GetQuotesButton({ modalView, errorText, setModalView, fetchQuotes }: GetQuotesButtonProps) {
-  const [loading, setLoading] = useState<boolean>(false)
+export default function GetQuotesButton({ errorText, setModalView, fetchQuotes }: GetQuotesButtonProps) {
   const { address: account } = useAccount()
   const { t } = useTranslation()
 
-  const Next = useCallback(async () => {
-    if (modalView === CryptoFormView.Input) {
-      setLoading(true)
-      fetchQuotes().then(() => {
-        setTimeout(() => {
-          setLoading(false)
-          setModalView(CryptoFormView.Quote)
-        }, 1000)
-      })
-    } else {
-      // to-do
-    }
-  }, [modalView, setModalView, fetchQuotes])
+  const { mutate, isLoading } = useMutation(fetchQuotes, {
+    onSuccess: () => {
+      setModalView(CryptoFormView.Quote)
+    },
+  })
+  const next = useCallback(() => mutate(), [mutate])
```

Comment:
Why don't we use `mutate` directly?

Reply:
If you want to rename the variable you could always do it on object destructuring like this

```typescript
const {
  mutate: next,
  isLoading,
} = useMutation(fetchQuotes, {
    onSuccess: () => {
      setModalView(CryptoFormView.Quote)
    },
  })
```

Reply:
lol thats fine. i actually did originally just call mutate, but then changed it as i thought u might say not to pass mutate directly. haah cool reverting that now